src/commands/auth.ts

```diff
@@ -78,84 +78,145 @@ const preserveCredentials = async (
   });
 };
 
+type TokenSetContents = {
+  user_id: string;
+} & TokenSet;
+
+const isValidTokenSet = (tokenSet: TokenSet): boolean => {
+  // Only require refresh_token for refresh attempts
+  return !!tokenSet.refresh_token;
```

Comment:
Not sure I understand this comment. `isValidTokenSet` is called at the beginning of `handleExistingToken`, so we're always checking if `refresh_token` is present, not just when refreshing?